libvast/src/system/index.cpp

```diff
@@ -284,7 +284,8 @@ void index_state::reset_active_partition() {
   // Persist meta data and the state of all INDEXER actors when the active
   // partition gets replaced becomes full.
   if (active != nullptr) {
-    stage->out().unregister(active.get());
+    if (!stage->out().unregister(active.get()))
+      VAST_ERROR(self, "tried to unregister unknown partition");
```

Comment:
How can this be a system error? I looks like a programming error if this occurs where an assertion would be in place. Otherwise there needs to be a clear comment why this internal logic should be propagated to the user.

Reply:
Correctly observed. It has been changed into an assertion.